docs/shared/guides/nx-and-ts.md

```diff
@@ -0,0 +1,151 @@
+# Nx and TypeScript
+
+> The build system for TypeScript that TypeScript deserves
+
+While you can use Nx with plain JavaScript (using the `--js` flag), Nx has always had first class support for TypeScript. It is the default when you scaffold a new workspace or generate new projects using the native Nx plugins.
+
+To provide even better TypeScript support we introduced the `@nrwl/js` with the intention of providing the **best possible developer experience for the TypeScript ecosystem**.
+
+<iframe width="560" height="315" src="https://www.youtube.com/embed/-OmQ-PaSY5M" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture; fullscreen"></iframe>
+
+`@nrwl/js` is particularly useful if you want to
+
+- Create framework agnostic, just plain TypeScript libraries within an existing Nx workspace (say to use in your React, Node or Angular app)
+- Publish pure TypeScript packages to NPM
+
+To get started with TypeScript packages in Nx, either add the `@nrwl/js` package to an existing Nx workspace or generate a new Nx workspace using the `--preset=ts` preset.
+
+```bash
+create-nx-workspace happynrwl -–preset=ts
+```
+
+Generating a new workspace creates a lightweight setup with a `packages` and `tools` folder. The `tools` folder is where you can add monorepo specific scripts and custom Nx generators, the `packages` folder is where all our TS based libs will live.
+
+```treeview
+happynrwl/
+├── packages/
+├── tools/
+├── workspace.json
+├── nx.json
+├── package.json
+└── tsconfig.base.json
+```
+
+As with most of native Nx plugins, `@nrwl/js` comes with a set of generators to quickly scaffold new TypeScript libraries as well as TypeScript applications. Let’s have a look at an example.
+
+## Create a new TypeScript based library
+
+We can use Nx Console and choose the `@nrwl/js:library` generator or directly use the following command:
+
+```bash
+nx generate @nrwl/js:library --name=hello-tsc --buildable
+```
+
+This creates a new library in the `packages/hello-tsc` folder that already comes with both, ESLint and Jest set up and ready to use.
+
+You can run `nx lint hello-tsc` to run linting or `nx test hello-tsc` to run Jest tests.
+
+Note, by passing the `--buildable` flag, our library can be built.
+
+```bash
+nx build hello-tsc
+```
+
+The output is placed into the `dist/packages/hello-tsc` by default.
+
+## Create a TypeScript based application
```

Comment:
I'd also show importing a library and say libraries and apps can import libraries